File: poc-kronos-frontend/src/model/curso/curso-documento/CursoDocumento.js
class CursoDocumento {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.original = dados ? dados.original : null;
    this.autenticacao = dados ? dados.autenticacao : null;
    this.numeroCopia = dados ? dados.numeroCopia : null;
    this.idTipoDocumento = dados ? dados.idTipoDocumento : null;
    this.nomeTipoDocumento = dados ? dados.nomeTipoDocumento : null;
    this.idCurso = dados ? dados.idCurso : null;
  }

  static validations(required, numeric) {
    return {
      numeroCopia: { required, numeric },
      idTipoDocumento: { required },
      idCurso: { required }
    };
  }

  static validationMessages() {
    return {
      numeroCopia: {
        required: "Número de Cópia é obrigatório",
        numeric: "Somente número"
      },
      idTipoDocumento: {
        required: "Tipo Documento é obrigatório"
      },
      idCurso: {
        required: "Curso é obrigatório"
      }
    };
  }
}
export default CursoDocumento;

File: src/main/java/br/com/kronos/backend/aplicacao/curso/CursoDocumento.java
package br.com.kronos.backend.aplicacao.curso;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import lombok.Builder;
import lombok.Getter;

@Getter
public class CursoDocumento {
    
    private long id;
    private boolean original;
    private boolean autenticacao;
    private Integer numeroCopia;
    private Long idCurso;
    private Integer idTipoDocumento;

	@Builder
	public CursoDocumento(long id,  boolean original, boolean autenticacao, int numeroCopia, Long idCurso, int idTipoDocumento) {

		super();
        this.id = id;
        this.original = original;
        this.autenticacao = autenticacao;
        this.numeroCopia = exigirNaoNulo(numeroCopia, "numeroCopia é obrigatório");
        this.idCurso = exigirNaoNulo(idCurso, "idCurso é obrigatório"); 
        this.idTipoDocumento = exigirNaoNulo(idTipoDocumento, "idTipoDocumento é obrigatório");  
	}
}
